### PR TITLE
Add SW registration template injection

### PR DESCRIPTION
**Pull request by ragingwind**
Solved https://github.com/vuejs-templates/pwa/issues/12

### CODE REVIEW

**Comment by HugoCrd** [comment]
Hi!

Thanks for this fix :) 

There's a small typo here: https://github.com/vuejs-templates/pwa/pull/21/files#diff-6648dbe228d17a7aab49fbc78aee89eaR74
`servieWorkerLoader` => `serviceWorkerLoader`

Best,

**Comment by ragingwind** [comment]
@HugoCrd Oh my bad, Thanks a lot. I fixed

File: template/build/webpack.prod.conf.js [hunk 1]
```diff
@@ -14,6 +15,9 @@ var env = {{#if_or unit e2e}}process.env.NODE_ENV === 'testing'
   ? require('../config/test.env')
   : {{/if_or}}config.build.env
 
+var serviceWorkerLoader = `<script>${fs.readFileSync(path.join(__dirname,
```
Comment (addyosmani) [inline template/build/webpack.prod.conf.js]: As this is only referenced once, shall we just inline it directly in `serviceWorkerLoader: <..>` lower down?

Comment (ragingwind) [inline template/build/webpack.prod.conf.js]: @addyosmani done and test in dev mode

File: template/build/webpack.dev.conf.js [hunk 2]
```diff
@@ -28,7 +28,8 @@ module.exports = merge(baseWebpackConfig, {
     new HtmlWebpackPlugin({
       filename: 'index.html',
       template: 'index.html',
-      inject: true
+      inject: true,
+      serviceWorkerLoader: ''
```
Comment (jeffposnick) [inline template/build/webpack.dev.conf.js]: An alternative to loading nothing would be to load a no-op service worker file, like https://github.com/facebookincubator/create-react-app/pull/2276/commits/8b426ad9a9c71d5113baa367bb0d25ff0bce0567

That works around an issue where someone might accidentally start up their prod environment on the port that they normally use for dev on localhost, causing the prod service worker to stick around indefinitely (see https://github.com/facebookincubator/create-react-app/issues/2272).

Comment (ragingwind) [inline template/build/webpack.dev.conf.js]: I will update for this

**Comment by ragingwind** [comment]
@jeffposnick @addyosmani PTAL. I've updated with changing of name in sw files to keep consistence in other build files

**Comment by jeffposnick** [comment]
👍

**Comment by addyosmani** [comment]
👍

**Pull request merged**